### PR TITLE
feat: F6 keyboard navigation between editor panels

### PR DESCRIPTION
**Pull request by Tristan578**
## Summary
- F6/Shift+F6 cycles focus between editor regions: Sidebar → Scene Hierarchy → Canvas → Inspector/Right Panel
- Standard IDE pattern (VS Code, IntelliJ) for pane navigation
- Added `data-editor-region` attributes for programmatic focus targeting
- Focus-visible ring indicators on all focusable regions
- `<main>` landmark with `aria-label` for screen reader navigation
- Disabled in compact/mobile mode where regions are inside conditional drawers
- Added `layout.mode` to `handleGlobalKeyDown` dependency array (stale closure fix)

Closes #8256

## Test plan
- [x] `regionFocusCycling.test.ts` — 4 tests: forward, backward, wrap-around, child-of-region
- [x] `EditorLayout.test.tsx` — F6 focus to sidebar, F6 disabled in compact mode
- [x] `keybindings.test.ts` — 28 existing tests pass with new entries
- [x] Lint: 0 warnings
- [x] TypeScript: no errors
- [ ] E2E: Tab from hierarchy to canvas, Delete entity (blocked on WASM — #8255)

### CODE REVIEW

File: web/src/components/editor/EditorLayout.tsx [hunk 1]
```diff
@@ -447,6 +447,21 @@ export function EditorLayout() {
...
+        const regions = ['sidebar', 'hierarchy', 'canvas', 'right-panel'] as const;
+        const targetEl = e.target instanceof HTMLElement ? e.target : null;
+        const current = targetEl?.closest('[data-editor-region]')?.getAttribute('data-editor-region');
+        const currentIdx = current ? regions.indexOf(current as typeof regions[number]) : -1;
+        const direction = e.shiftKey ? -1 : 1;
+        const nextIdx = (currentIdx + direction + regions.length) % regions.length;
+        const nextEl = document.querySelector(`[data-editor-region="${regions[nextIdx]}"]`) as HTMLElement | null;
+        nextEl?.focus();
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/components/editor/EditorLayout.tsx:473]: Fixed in `905f92bc`. Added `!isInput` guard to the F6 condition at line 453 so it skips when focus is in INPUT, TEXTAREA, or contentEditable elements.

File: web/src/components/editor/EditorLayout.tsx [hunk 2]
```diff
@@ -447,6 +447,21 @@ export function EditorLayout() {
...
+      // F6 / Shift+F6: Cycle focus between editor regions (standard IDE pattern)
+      // Disabled in compact mode where regions are inside conditional drawers
+      if (e.key === 'F6' && layout.mode !== 'compact') {
+        e.preventDefault();
+        const regions = ['sidebar', 'hierarchy', 'canvas', 'right-panel'] as const;
+        const targetEl = e.target instanceof HTMLElement ? e.target : null;
+        const current = targetEl?.closest('[data-editor-region]')?.getAttribute('data-editor-region');
+        const currentIdx = current ? regions.indexOf(current as typeof regions[number]) : -1;
+        const direction = e.shiftKey ? -1 : 1;
+        const nextIdx = (currentIdx + direction + regions.length) % regions.length;
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/components/editor/EditorLayout.tsx]: Fixed in `905f92bc`. Added special case for `currentIdx === -1`: forward starts at index 0 (sidebar), backward starts at `regions.length - 1` (right-panel). Regression test added in regionFocusCycling.test.ts.

File: web/src/components/editor/__tests__/EditorLayout.test.tsx [hunk 3]
```diff
@@ -396,4 +396,37 @@ describe('EditorLayout', () => {
...
+
+    const prevActive = document.activeElement;
+
+    act(() => {
+      document.dispatchEvent(new KeyboardEvent('keydown', { key: 'F6', bubbles: true }));
+    });
+
+    // Focus should not have moved
+    expect(document.activeElement).toBe(prevActive);
+  });
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/components/editor/__tests__/EditorLayout.test.tsx:431]: Fixed in `905f92bc`. Added test "cycles to right-panel when nothing is focused (Shift+F6 backward)" in regionFocusCycling.test.ts line 58.

File: web/src/components/editor/__tests__/regionFocusCycling.test.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+ * Extracted here so we can test it without rendering the full component tree.
+ */
+function cycleRegion(direction: 1 | -1): void {
+  const targetEl = document.activeElement instanceof HTMLElement ? document.activeElement : null;
+  const current = targetEl?.closest('[data-editor-region]')?.getAttribute('data-editor-region');
+  const currentIdx = current ? REGIONS.indexOf(current as typeof REGIONS[number]) : -1;
+  const nextIdx = (currentIdx + direction + REGIONS.length) % REGIONS.length;
+  const nextEl = document.querySelector(`[data-editor-region="${REGIONS[nextIdx]}"]`) as HTMLElement | null;
+  nextEl?.focus();
+}
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/components/editor/__tests__/regionFocusCycling.test.ts:30]: By design. All panel shortcuts in EditorLayout (Ctrl+K, F1, ?, Alt+T) are hardcoded with a corresponding display-only entry in keybindings.ts. Extracting a shared module for a 6-line function would add indirection without reducing drift risk — the test intentionally validates the algorithm in isolation from the React tree.

File: web/src/lib/workspace/keybindings.ts [hunk 5]
```diff
@@ -55,6 +55,8 @@ export const DEFAULT_BINDINGS: KeyBinding[] = [
   { action: 'open-docs', label: 'Open docs', category: 'Panels', defaultKey: 'F1', customKey: null },
   { action: 'show-shortcuts', label: 'Show shortcuts', category: 'Panels', defaultKey: '?', customKey: null },
   { action: 'toggle-taskboard', label: 'Toggle taskboard', category: 'Panels', defaultKey: 'Alt+T', customKey: null },
+  { action: 'cycle-region', label: 'Cycle editor region', category: 'Panels', defaultKey: 'F6', customKey: null },
+  { action: 'cycle-region-back', label: 'Cycle region (reverse)', category: 'Panels', defaultKey: 'Shift+F6', customKey: null },
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/lib/workspace/keybindings.ts:59]: By design — consistent with all other panel shortcuts. `toggle-chat` (Ctrl+K), `open-docs` (F1), `show-shortcuts` (?), and `toggle-taskboard` (Alt+T) are all hardcoded handlers with display-only registry entries. The entries exist so the Shortcuts Panel shows users what F6/Shift+F6 do. Wiring customizable bindings for region cycling is out of scope for this PR.

**Comment by Tristan578** [comment]
## Review Comment Resolution

All 5 review comments addressed in `905f92bc`.

| File | Reviewer | Resolution |
|------|----------|------------|
| EditorLayout.tsx#463 | @sentry | Fixed — added `!isInput` guard so F6 skips text inputs |
| EditorLayout.tsx#459 | @Copilot | Fixed — Shift+F6 from unfocused now starts at last region |
| EditorLayout.test.tsx#431 | @Copilot | Fixed — added Shift+F6 unfocused test |
| regionFocusCycling.test.ts#27 | @Copilot | By design — consistent with all other panel shortcuts |
| keybindings.ts#59 | @Copilot | By design — display-only entries match Ctrl+K/F1/?/Alt+T pattern |

File: web/src/components/editor/EditorLayout.tsx [hunk 6]
```diff
@@ -447,6 +447,25 @@ export function EditorLayout() {
...
+        const targetEl = e.target instanceof HTMLElement ? e.target : null;
+        const current = targetEl?.closest('[data-editor-region]')?.getAttribute('data-editor-region');
+        const currentIdx = current ? regions.indexOf(current as typeof regions[number]) : -1;
+        const direction = e.shiftKey ? -1 : 1;
+        // When unfocused (currentIdx=-1), forward starts at 0, backward at last
+        const nextIdx = currentIdx === -1
+          ? (direction === 1 ? 0 : regions.length - 1)
+          : (currentIdx + direction + regions.length) % regions.length;
+        const nextEl = document.querySelector(`[data-editor-region="${regions[nextIdx]}"]`) as HTMLElement | null;
+        nextEl?.focus();
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/components/editor/EditorLayout.tsx:471]: Fixed in `d8a9fd28`. F6 handler now filters `allRegions` to only those with a matching `[data-editor-region]` element in the DOM. In desktop mode, `right-panel` is managed by Dockview (WorkspaceProvider) and is not in the region list. Cycling wraps correctly across sidebar → hierarchy → canvas → sidebar.

File: web/src/components/editor/EditorLayout.tsx [hunk 7]
```diff
@@ -447,6 +447,31 @@ export function EditorLayout() {
...
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/components/editor/EditorLayout.tsx:471]: By design. The handler already filters \`allRegions\` to only DOM-mounted elements (line 457: \`allRegions.filter(r => document.querySelector(...))\`), so in desktop mode where right-panel is unmounted, cycling correctly wraps across sidebar → hierarchy → canvas → sidebar. The \`allRegions\` array includes right-panel for forward compatibility when Dockview panels get the attribute.

File: web/src/components/editor/__tests__/regionFocusCycling.test.ts [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+function cycleRegion(direction: 1 | -1): void {
+  const targetEl = document.activeElement instanceof HTMLElement ? document.activeElement : null;
+  const current = targetEl?.closest('[data-editor-region]')?.getAttribute('data-editor-region');
+  const currentIdx = current ? REGIONS.indexOf(current as typeof REGIONS[number]) : -1;
+  // When unfocused (currentIdx=-1), forward starts at 0, backward at last
+  const nextIdx = currentIdx === -1
+    ? (direction === 1 ? 0 : REGIONS.length - 1)
+    : (currentIdx + direction + REGIONS.length) % REGIONS.length;
+  const nextEl = document.querySelector(`[data-editor-region="${REGIONS[nextIdx]}"]`) as HTMLElement | null;
+  nextEl?.focus();
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply web/src/components/editor/__tests__/regionFocusCycling.test.ts:29]: By design. The DOM-level test creates all 4 regions in the test DOM (lines 42-45) to validate the cycling algorithm works for any number of regions. It tests the math (modular arithmetic, direction, unfocused edge cases) in isolation from the React component tree. The EditorLayout.test.tsx tests verify the integration with actual mounted regions.

File: .changeset/keyboard-nav-f6-cycling.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+---
+'web': patch
```
[the remark this answers is not in the document]
Comment (Tristan578) [reply .changeset/keyboard-nav-f6-cycling.md]: Fixed in \`203356ed\`. Changed to double-quoted package name to match the documented changeset format.
